### ISSUE TITLE
Find a New Way to Update Popular and Trending Pages

### ISSUE THREAD

**Issue by harshkhandeparkar** [metakgp/metakgp-wiki#111]
- The `UA-XXXXXX-XX` google analytics id is deprecated and hence the `googleAnalytics` extension no longer works.
- Need to find a way to use the new "head" tags of Google Analytics and adapt the jobs script accordingly.

**Comment by shikharish** [comment]
What is the difference between the Popular and Trending sections?

**Comment by harshkhandeparkar** [comment]
Popular is the list of all-time most visited pages and Trending is list of pages with recent spikes in traffic.

**Comment by shikharish** [comment]
Does the popular section use an extension? 

In `update_top_trending.sh`, it uses a script `pwb.py` but I cannot find it anywhere.
![image](https://github.com/metakgp/metakgp-wiki/assets/139053348/4e861c38-0c9e-4cb6-a011-9e4f0ccd4074)

**Comment by harshkhandeparkar** [comment]
>Does the popular section use an extension?

I don't think so. Just the newest pages section uses an extension.

> In update_top_trending.sh, it uses a script pwb.py but I cannot find it anywhere.

pwb.py just calls PyWikiBot. The `login` and `updatestatistics` are the main scripts. `pwb.py` will be installed on build.

**Comment by shikharish** [comment]
Pls assign me this issue.

**Comment by harshkhandeparkar** [comment]
Go ahead!

**Comment by shikharish** [comment]
Popular and Trending pages both use the same script. Both are broken. Maybe edit the issue title to add Popular pages too.

**Comment by shikharish** [comment]
@harshkhandeparkar We can use another Google Analytics extension to fix this, but that raises privacy concerns. This is from the extension page:
>The GoogleAnalyticsMetrics extension may not be appropriate for your site if you have data security and privacy concerns. The extensions inserts JavaScript code in each page served, and the code causes the client to contact Google services to process and aggregate data. Your site's users will be subjected to tracking and data mining regardless of the [Do Not Track](https://en.wikipedia.org/wiki/Do_Not_Track) or [Global Privacy Control](https://en.wikipedia.org/wiki/Global_Privacy_Control) browser settings.

 Should we use this?

**Comment by harshkhandeparkar** [comment]
Ideally not. Is there any extension that tracks page views without using any external service?

**Comment by shikharish** [comment]
> Ideally not. Is there any extension that tracks page views without using any external service?

Not one I could find. There is a TopTenPages extension but that only counts unique page views and arranges acc to all time.

**Comment by harshkhandeparkar** [comment]
> > Ideally not. Is there any extension that tracks page views without using any external service?
> 
> Not one I could find. There is a TopTenPages extension but that only counts unique page views and arranges acc to all time.

Is there any way to expose an option to the user to disable google analytics?

**Comment by shikharish** [comment]
Could not find any.

**Comment by harshkhandeparkar** [comment]
Can google analytics be configured to only record anonymous view count and nothing else?

**Comment by shikharish** [comment]
Yes we can specify which metrics we want the extension to record.

**Comment by harshkhandeparkar** [comment]
Then I think we can go ahead with this.

**Comment by shikharish** [comment]
Alright lets do this!

**Comment by shikharish** [comment]
@harshkhandeparkar Turns out it is not possible to configure which metrics we want to send in Google Analytics. It would include [many](https://developers.google.com/analytics/devguides/reporting/data/v1/api-schema) metrics which are neither needed nor privacy focused. I say we remove the Trending pages section completely. We can replace the Popular pages with a simple extension like [HitCounters](https://www.mediawiki.org/wiki/Extension:HitCounters).

**Comment by harshkhandeparkar** [comment]
> @harshkhandeparkar Turns out it is not possible to configure which metrics we want to send in Google Analytics. It would include [many](https://developers.google.com/analytics/devguides/reporting/data/v1/api-schema) metrics which are neither needed nor privacy focused. I say we remove the Trending pages section completely. We can replace the Popular pages with a simple extension like [HitCounters](https://www.mediawiki.org/wiki/Extension:HitCounters).

Would it be possible to integrate the HitCounters page on the main page?

**Comment by shikharish** [comment]
Yes, we could make a template from it.

**Comment by harshkhandeparkar** [comment]
> Yes, we could make a template from it.

Nice. Go ahead with this.

**Comment by harshkhandeparkar** [comment]
@shikharish any update?

**Comment by harshkhandeparkar** [comment]
NOTE: The sidebar is no longer shown with Vector 2022 skin.
